Userland/Utilities/uniq.cpp

```diff
@@ -5,75 +5,122 @@
  */
 
 #include <AK/RefPtr.h>
+#include <AK/StringView.h>
 #include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
 #include <LibCore/System.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
 #include <unistd.h>
 
-struct linebuf {
-    char* buf = NULL;
-    size_t len = 0;
-};
-
-static FILE* get_stream(char const* filepath, char const* perms)
+static ErrorOr<void> write_line_content(StringView line, size_t count, bool show_duplicates, bool print_count, Core::Stream::File& outfile)
 {
-    FILE* ret;
-
-    if (filepath == nullptr) {
-        if (perms[0] == 'r')
-            return stdin;
-        return stdout;
+    if (!show_duplicates || count > 0) {
+        if (print_count)
+            TRY(outfile.write(String::formatted("{} {}\n", count + 1, line).bytes()));
+        else
+            TRY(outfile.write(String::formatted("{}\n", line).bytes()));
     }
+    return {};
+}
 
-    ret = fopen(filepath, perms);
-    if (ret == nullptr) {
-        perror("fopen");
-        exit(1);
+static StringView skip(StringView line, unsigned int char_skip_count, unsigned int field_skip_count)
+{
+    line = line.trim("\n"sv);
+    if (field_skip_count) {
+        bool in_field = false;
+        int field_index = 0;
+        unsigned int current_field = 0;
+        for (size_t i = 0; i < line.length(); i++) {
+            char c = line[i];
+            if (c == ' ' || c == '\t' || c == '\v' || c == '\f' || c == '\r') {
+                in_field = false;
+                field_index = i;
+                if (++current_field > field_skip_count)
+                    break;
+            } else if (!in_field) {
+                in_field = true;
+            }
+        }
+        line = line.substring_view(field_index);
     }
+    char_skip_count = min(char_skip_count, line.length());
+    return line.substring_view(char_skip_count);
+}
 
-    return ret;
+// Borrowed from cmp.cpp, there should probably be a simpler way
+static ErrorOr<NonnullOwnPtr<Core::Stream::BufferedFile>> open_file_or_stdin(String const& filename)
+{
+    OwnPtr<Core::Stream::File> file;
+    if (filename.is_empty() || filename == "-")
+        file = TRY(Core::Stream::File::adopt_fd(STDIN_FILENO, Core::Stream::OpenMode::Read));
+    else
+        file = TRY(Core::Stream::File::open(filename, Core::Stream::OpenMode::Read));
+    return TRY(Core::Stream::BufferedFile::create(file.release_nonnull()));
 }
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
     TRY(Core::System::pledge("stdio rpath wpath cpath"));
 
-    char const* inpath = nullptr;
-    char const* outpath = nullptr;
+    StringView inpath;
+    StringView outpath;
+    bool duplicates = false;
+    bool unique = false;
```

Comment:
These variable names are a little misleading. `duplicates` doesn't mean "print duplicates", it means "*only* print duplicates". I'd probably choose something more like:
```suggestion
    bool duplicates_only = false;
    bool unique_only = false;
```
Or even `print_duplicates_only` to be really clear. We're big fans of really clear names. :^)

Reply:
bool flag_that_makes_the_program_only_print_duplicates_rather_than_the_default_one_being_printing_only_unique_ones = false